pkg/kafka_client/client.go

```diff
@@ -13,12 +13,26 @@ const MAX_EARLIEST int64 = 100
 
 type Options struct {
 	BootstrapServers string `json:"bootstrapServers"`
+	SecurityProtocol string `json:"securityProtocol"`
+	SaslMechanisms   string `json:"saslMechanisms"`
+	SaslUsername     string `json:"saslUsername"`
+	SaslPassword     string `json:"saslPassword"`
+	// TODO: If Debug is before HealthcheckTimeout, then json.Unmarshall
+	// silently fails to parse the timeout from the s.JSONData.  Figure out why.
+	HealthcheckTimeout int32  `json:"healthcheckTimeout"`
+	Debug              string `json:"debug"`
```

Comment:
This cost me a good chunk of my free weekend time.  If I swap the order of HealthcheckTimeout and Debug here, the json.Unmarshall() in getDatasourceSettings() in plugin.go just silently skips setting a value.  I tried changing the value type to string all the way through with a strconv.Atoi() at the end, still fails.  I think it is something in the Debug logic, but Go is not a main language of mine and I've never touched any Node code before.

Reply:
It's OK. I will check that out and fix it.
Thanks.